insights/client/connection.py

```diff
@@ -616,92 +661,87 @@ def do_group(self):
-            # unregistered_at = null, means this machine IS registered
-            return True
-        else:
+            # so behavior doesn't change too much, consider this "unreachable"
+            raise RuntimeError('Could not reach the Insights API.')
+
+        # check the 'unregistered_at' key of the response
+        unreg_status = sysdata.get('unregistered_at', None)
+        # set the global account number
+        self.config.account_number = sysdata.get('account_number', 'undefined')
+
+        if unreg_status:
             # machine has been unregistered, this is a timestamp
             return unreg_status
+        else:
+            # unregistered_at = null, means this machine IS registered
+            return True
 
     def _fetch_system_by_machine_id(self):
         '''
         Get a system by machine ID
         Returns
             dict    system exists in inventory
-            False   system does not exist in inventory
-            None    error connection or parsing response
+            None    system does not exist in inventory
         '''
         machine_id = generate_machine_id()
-        try:
-            # [circus music]
-            if self.config.legacy_upload:
-                url = self.base_url + '/platform/inventory/v1/hosts?insights_id=' + machine_id
-            else:
-                url = self.inventory_url + '/hosts?insights_id=' + machine_id
-            logger.log(NETWORK, "GET %s", url)
-            res = self.session.get(url, timeout=self.config.http_timeout)
-        except REQUEST_FAILED_EXCEPTIONS as e:
-            _api_request_failed(e)
+        # [circus music]
+        if self.config.legacy_upload:
+            url = self.base_url + '/platform/inventory/v1/hosts?insights_id=' + machine_id
+        else:
+            url = self.inventory_url + '/hosts?insights_id=' + machine_id
+        res = self.get(url)
+        if not res:
             return None
         try:
-            if (self.handle_fail_rcs(res)):
-                return None
             res_json = json.loads(res.content)
         except ValueError as e:
             logger.error(e)
             logger.error('Could not parse response body.')
             return None
         if res_json['total'] == 0:
             logger.debug('No hosts found with machine ID: %s', machine_id)
-            return False
+            return None
         return res_json['results']
 
     def api_registration_check(self):
         '''
-            Reach out to the inventory API to check
-            whether a machine exists.
+        Reach out to the inventory API to check
+        whether a machine exists.
 
-            Returns
-                True    system exists in inventory
-                False   system does not exist in inventory
-                None    error connection or parsing response
+        Returns
+            True    system exists in inventory
+            False   system does not exist in inventory
```

Comment:
This can still return _None_ in case of legacy upload. [__legacy_api_registration_check_](https://github.com/RedHatInsights/insights-core/blob/bb04fb85035fbb0eb8bab72486328bb2ad8cc9d1/insights/client/connection.py#L661) [does so](https://github.com/RedHatInsights/insights-core/blob/bb04fb85035fbb0eb8bab72486328bb2ad8cc9d1/insights/client/connection.py#L681) on error 404.